src/citrine/resources/dataset.py

```diff
@@ -393,14 +397,18 @@ class DatasetCollection(Collection[Dataset]):
 
     """
 
-    _path_template = 'projects/{project_id}/datasets'
+    _path_template = None
     _individual_key = None
     _collection_key = None
     _resource = Dataset
 
-    def __init__(self, project_id: UUID, session: Session):
-        self.project_id = project_id
+    def __init__(self, project_or_team: Either[UUID, UUID], session: Session):
+        self.project_or_team = project_or_team
```

Comment:
I thought you said that was fine here because nobody was using that syntax.

https://github.com/CitrineInformatics/citrine-python/pull/927#discussion_r1561573166

By that moment I was thinking on creating a new collection for the team datasets to not alter the constructor. What would be the best for you? Do you want me to internalize that flag telling me what kind of collection is and only add `team_id` to the constructor? That way I'd leave `project_id` as is. I can then verify that this is called with one and only one of the two options and fail with an exception if it is not. Does that sound better?

Reply:
Right when I said that the common pattern was
```
citrine = Citrine()
team = citrine.teams.get('xxx')
project = team.projects.get('xxx')
dataset = project.datasets.get('xxx')
```
I think I was saying that having the above functionally short circuit to
```
citrine = Citrine()
team = citrine.teams.get('xxx')
dataset = team.datasets.get('xxx')
```
by having `project.datasets.get('xxx')` be functionally equivalent to (pseudocode) `Team(project.team_id).datasets.get('xxx')`, then we support both the existing interface and the new pattern.  This could also be implemented either through explicitly creating a Team object, or by simply changing the route that the existing endpoint hits.  

We should also ask the question of if we want to deprecate the entire `Project.datasets` family of accessors, or if we think they are convenient enough that we want to maintain the redirect for the sake of ergonomics.  As a point in favor, if I were doing AI modeling (operating in a `project` context), I could definitely gain benefit in having a single command access to all potential dataset, as opposed to requiring holding onto a second `team` variable or having to chain methods.

> By that moment I was thinking on creating a new collection for the team datasets to not alter the constructor. What would be the best for you? Do you want me to internalize that flag telling me what kind of collection is and only add team_id to the constructor? That way I'd leave project_id as is. I can then verify that this is called with one and only one of the two options and fail with an exception if it is not. Does that sound better?

Why would we create a new collection for TeamDatasets instead of just adding a `team_id` to the existing Dataset collection?  The way those internal identifiers are handled is frankly a mess right now, but at least that keeps the paradigm consistent until we decide on a general refactor.  In Python, it's perfectly fine to use `None` as an indicator of object state.  Those are also really not user facing -- like I said, it need refactoring.

Reply:
I was choosing to create a new collection to prevent a lot of branching in the code as the project-based code will be killed (here I'm talking about calling the project-based URLs or any similar behavior). No matter if the datasets collection in the project is a proxy for team one, there is no reason for maintaining a project_id in the dataset collection, or is it? On whether we should have this proxy or not, I think that's more a call for @jspeerless or you, I think it could be beneficial for ergonomics if it doesn't make people confused about ownership. I'll not disagree if we think it is the best. Let me push a new version of this. But I still believe the project_id should be removed from the collection at a given point of time being that a breaking change if someone is not using the patterns you are referring to.